v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/endtoend/EndToEndTestingITBase.java

```diff
@@ -0,0 +1,364 @@
+import org.apache.beam.it.gcp.datastream.DatastreamResourceManager;
+import org.apache.beam.it.gcp.datastream.JDBCSource;
+import org.apache.beam.it.gcp.datastream.MySQLSource;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.beam.it.gcp.spanner.SpannerResourceManager;
+import org.apache.beam.it.gcp.storage.GcsResourceManager;
+import org.apache.beam.it.jdbc.JDBCResourceManager;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class EndToEndTestingITBase extends TemplateTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(EndToEndTestingITBase.class);
+  private static FlexTemplateDataflowJobResourceManager flexTemplateDataflowJobResourceManager;
+  public DatastreamResourceManager datastreamResourceManager;
+  protected JDBCSource jdbcSource;
+
+  protected SpannerResourceManager createSpannerDatabase(String spannerSchemaFile)
+      throws IOException {
+    SpannerResourceManager spannerResourceManager =
+        SpannerResourceManager.builder("e2e-main-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build();
+
+    String ddl;
+    try (InputStream inputStream =
+        Thread.currentThread().getContextClassLoader().getResourceAsStream(spannerSchemaFile)) {
+      if (inputStream == null) {
+        throw new FileNotFoundException("Resource file not found: " + spannerSchemaFile);
+      }
+      try (BufferedReader reader =
+          new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+        ddl = reader.lines().collect(Collectors.joining("\n"));
+      }
+    }
+
+    if (ddl.isBlank()) {
+      throw new IllegalStateException("DDL file is empty: " + spannerSchemaFile);
+    }
+
+    String[] ddls = ddl.trim().split(";");
+    for (String d : ddls) {
+      d = d.trim();
+      if (!d.isEmpty()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    return spannerResourceManager;
+  }
+
+  protected SpannerResourceManager createSpannerMetadataDatabase() {
+    SpannerResourceManager spannerMetadataResourceManager =
+        SpannerResourceManager.builder("e2e-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build();
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
+    return spannerMetadataResourceManager;
+  }
+
+  public PubsubResourceManager setUpPubSubResourceManager() throws IOException {
+    return PubsubResourceManager.builder(testName, PROJECT, credentialsProvider).build();
+  }
+
+  // createPubsubResources generates pubsub topic, subscription and notification for migration.
+  // It can be run in different modes based on type of migration.
+  public SubscriptionName createPubsubResources(
```

Comment:
Can you add allowed values of mode in the doc comment?

Reply:
Sure